### PR TITLE
Reverse geocoding

### PR DESCRIPTION
**Pull request by Ef55**

### CODE REVIEW

**Review by justagoodguy-omega — APPROVED** [review]
Great code what can I say

File: app/src/main/java/com/github/onedirection/geocoding/NominatimGeocoding.java [hunk 1]
```diff
@@ -68,8 +83,32 @@ private static String generateRequestURL(String query){
...
+        );
+        requestQueue.add(request);
+        return result;
+    }
+
     @Override
     public CompletableFuture<NamedCoordinates> getBestNamedCoordinates(String locationName) {
-        return Monads.flatten(sendRequest(locationName).thenApply(NominatimGeocoding::parseResult));
+        return Monads.flatten(
+                sendArrayRequest(generateSearchRequestURL(locationName)).thenApply(r -> parseResult(r, 0))
```
[the remark this answers is not in the document]
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/geocoding/NominatimGeocoding.java:104]: odersky

File: app/src/main/java/com/github/onedirection/geocoding/NominatimGeocoding.java [hunk 2]
```diff
@@ -36,25 +39,37 @@ public NominatimGeocoding(Context ctx){
...
+        }
+        catch(Exception e){
+            Log.d(LOGCAT_TAG, "Parsing failed");
+            return Optional.empty();
+        }
     }
 
-    private static Optional<NamedCoordinates> parseResult(JSONArray json){
-        Log.d(LOGCAT_TAG, "Parsing: " + json);
+    private static Optional<NamedCoordinates> parseResult(JSONArray json, int idx){
```
Comment (TheImpressiveDonut) [inline app/src/main/java/com/github/onedirection/geocoding/NominatimGeocoding.java:61]: You only call this private method with idx = 0, so idx is not really needed. But I suppose this method maybe used for other things in the future?

**Review by TheImpressiveDonut — COMMENTED** [review]
One question on a private method

File: app/src/main/java/com/github/onedirection/geocoding/NominatimGeocoding.java (hunk 2 above)
Comment (Ef55) [reply app/src/main/java/com/github/onedirection/geocoding/NominatimGeocoding.java:61]: This arg specifies which result to retrieve. Currently, I only use the first one (`idx = 0`), so it is indeed useless.
However, adding this arg didn't take much time (as you may have guessed), and leaving the choice to the user to select which result to use is something we might add later, so I already allowed that in this particular method.

**Review by TheImpressiveDonut — APPROVED** [review]

**Pull request merged**